openedx/adg/lms/student/analytics.py

```diff
@@ -0,0 +1,45 @@
+"""
+Analytics related functions for student app
+"""
+import json
+import logging
+
+import requests
+from django.conf import settings
+from rest_framework.status import HTTP_200_OK
+
+log = logging.getLogger(__name__)
+
+
+def remove_user_data_from_segment(user):
+    """
+    Remove user previous data & suppress new data from segment API
+
+    Arguments:
+        user (User): User to remove data
+
+    Return:
+        boolean: True if the request is successful
+    """
+    if getattr(settings, 'SEGMENT_BEARER_TOKEN', None) and getattr(settings, 'SEGMENT_WORKSPACE', None):
+        url = f'https://platform.segmentapis.com/v1beta/workspaces/{settings.SEGMENT_WORKSPACE}/regulations'
+
+        data = json.dumps({
+            'regulation_type': 'Suppress_With_Delete',
+            'attributes': {
+                'name': 'userId',
+                'values': [str(user.id)]
+            }
+        })
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {settings.SEGMENT_BEARER_TOKEN}'
+        }
+
+        response = requests.request('POST', url, headers=headers, data=data)
+        log.info(f'Deleting User {user.username} data from Segment API, response code: {response.status_code}')
+        return response.status_code == HTTP_200_OK
+
+    log.info(f'Segment Bearer Token or Workspace not found!')
```

Comment:
Perhaps a little more info.  
Cannot remove segment data for user `email` failed. Segment Bearer Token or Workspace not found!